msgpack/fallback.py

```diff
@@ -404,7 +405,7 @@ def _reserve(self, n):
             self._buffer += read_data
             remain_bytes -= len(read_data)
 
-        if len(self._buffer) < n + self._buff_i:
+        if len(self._buffer) < n + self._buff_i and raise_outofdata:
```

Comment:
This is slightly messy, but required to enable `read_bytes` with more than number of bytes available in the input, which cmsgpack supports.